gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2452,7 +2467,10 @@ public static void GET_SNAPSHOT_TXID(String[] txid) throws SQLException {
     } else {
       txid[0] = "null";
     }
-    tc.clearActiveTXState(false, true);
+    // tc commit will clear all the artifacts but will not commit actual txState
+    // that should be committed in COMMIT procedure
+    tc.resetActiveTXState(true);
+    TXManagerImpl.getOrCreateTXContext().clearTXState();
     TXManagerImpl.snapshotTxState.set(null);
```

Comment:
This is because conn is returned to the pool after each task and may be used by another unrelated task.